### PR TITLE
Add perf-dashboard object yaml files and Makefile

### PR DESCRIPTION
**Pull request by clarketm**
close https://github.com/istio/istio/issues/22356

- [x] External IP created: 35.244.130.96.
- [x] DNS A record setup: https://perf.dashboard.istio.io/.
- [x] GKE cluster created in `istio-testing`.
- [x] Provision TLS cert.
- [x] Convert cluster resources to object yaml files.
- [x] Add a `Makefile` with _targets_ to build/push image and deploy k8s files.
- [x] Update Django `ALLOWED_HOSTS` for GCLB health checks.
- [x] Create a _postsubmit_ job to auto-{push,deploy} on changes to `perf_dashboard/` directory. (https://github.com/istio/test-infra/pull/2529)

### CODE REVIEW

File: perf_dashboard/deploy/perf-dashboard/Chart.yaml [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+dependencies:
+- name: metrics-server
+  version: ^2.0.3
+  repository: https://kubernetes-charts.storage.googleapis.com/
+keywords:
+- istio
+- perf
+- perf-dashboard
+sources:
+- https://github.com/istio/tools
```
Comment (carolynhu) [inline perf_dashboard/deploy/perf-dashboard/Chart.yaml]: Sources could be: https://github.com/istio/tools/perf_dashboard instead?

File: perf_dashboard/deploy/perf-dashboard/values.yaml [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+replicas: 3
+minReplicas: 1
+CPUUtilization: 80
+
+fullnameOverride: perf-dashboard
+ipName: perf-dashboard
+domain: perf.dashboard.istio.io
+
+image: gcr.io/istio-testing/perf-dashboard
+version: latest
```
Comment (carolynhu) [inline perf_dashboard/deploy/perf-dashboard/values.yaml]: so this `latest` version is what you mentioned the timestamp version?

File: perf_dashboard/perf_dashboard/settings.py [hunk 3]
```diff
@@ -39,7 +40,13 @@
 # SECURITY WARNING: don't run with debug turned on in production!
 DEBUG = True
 
-ALLOWED_HOSTS = ['localhost', 'perf.dashboard.qualistio.org']
+ALLOWED_HOSTS = [
+    'localhost',
+    'perf.dashboard.qualistio.org',
```
Comment (carolynhu) [inline perf_dashboard/perf_dashboard/settings.py]: I'm wondering if we are using 'perf.dashboard.istio.io' do we still need this 'perf.dashboard.qualistio.org'?

**Comment by carolynhu** [comment]
also, a chart lint error, you may need to fix, thanks

File: perf_dashboard/deploy/perf-dashboard/values.yaml (hunk 2 above)
Comment (clarketm) [reply perf_dashboard/deploy/perf-dashboard/values.yaml]: I put that there as a _default_. When this is deployed the version is computed from the [date and sha](https://github.com/istio/tools/blob/73bf3fe96d70e0a6497065147ec26ce841b717d0/perf_dashboard/Makefile#L1). (e.g. `v20200321-67dfdb7`)

File: perf_dashboard/perf_dashboard/settings.py (hunk 3 above)
Comment (clarketm) [reply perf_dashboard/perf_dashboard/settings.py]: It depends on whether we want to continue to serve from `perf.dashboard.qualistio.org` for a short period of time for migration (and forward via a cname record)?

**Comment by clarketm** [comment]
> also, a chart lint error, you may need to fix, thanks

Thanks. I am not a helm expert, so I need to figure out how to fix this, then I will update.

File: perf_dashboard/perf_dashboard/settings.py (hunk 3 above)
Comment (carolynhu) [reply perf_dashboard/perf_dashboard/settings.py]: yeah, keep it there for now, after the 'perf.dashboard.istio.io' is stable, we no longer need it.

File: perf_dashboard/deploy/perf-dashboard/Chart.yaml (hunk 1 above)
Comment (clarketm) [reply perf_dashboard/deploy/perf-dashboard/Chart.yaml]: Updated.

**Comment by clarketm** [comment]
> > also, a chart lint error, you may need to fix, thanks
> 
> Thanks. I am not a helm expert, so I need to figure out how to fix this, then I will update.

Alright, I think all the linting issues are resolved now.

File: perf_dashboard/deploy/perf-dashboard/Chart.lock [hunk 4]
```diff
@@ -0,0 +1,6 @@
+dependencies:
+- name: metrics-server
```
Comment (howardjohn) [inline perf_dashboard/deploy/perf-dashboard/Chart.lock]: Why do we need this? GKE already has this deployed I think?

Comment (clarketm) [reply perf_dashboard/deploy/perf-dashboard/Chart.lock]: If that is the case, then we can remove this.

**Pull request merged**